File: mobile/android/app/src/main/java/com/tigerwallet/app/trading/P2PService.java
package com.tigerwallet.app.trading;

/**
 * P2P Trading Service - Android Implementation
 */

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class P2PService {
    
    public static class P2PAdvert {
        public String id;
        public String userId;
        public String username;
        public String avatar;
        public String side;
        public String token;
        public String fiatCurrency;
        public String paymentMethod;
        public double price;
        public double minAmount;
        public double maxAmount;
        public double availableAmount;
        public int ordersCompleted;
        public double completionRate;
        public double avgReleaseTime;
        public boolean isOnline;
    }
    
    public static class P2POrder {
        public String id;
        public String advertId;
        public String side;
        public String token;
        public String fiatCurrency;
        public String paymentMethod;
        public double price;
        public double amount;
        public double fiatAmount;
        public String status;
    }
    
    private static final String[] users = {"CryptoTrader1", "BitSeller", "FastTrade", "P2PPro"};
    private static final String[] avatars = {"🧑‍💼", "👨‍💻", "⚡", "🎯"};
    private static final String[] payments = {"Bank Transfer", "PayPal", "AliPay", "UPI"};
    private static final double[] basePrices = {1.0, 43250.0, 2280.0, 312.5};
    
    public List<P2PAdvert> getAdverts(String token, String fiatCurrency, String side) {
        List<P2PAdvert> adverts = new ArrayList<>();
        for (int i = 0; i < users.length; i++) {
            P2PAdvert advert = new P2PAdvert();
            advert.id = "advert_" + i;
            advert.userId = "user_" + i;
            advert.username = users[i];
            advert.avatar = avatars[i];
            advert.side = side != null ? side : (i % 2 == 0 ? "BUY" : "SELL");
            advert.token = token != null ? token : "USDT";
            advert.fiatCurrency = fiatCurrency != null ? fiatCurrency : "USD";
            advert.paymentMethod = payments[i];
            advert.price = basePrices[i % 4] * (1 + (Math.random() * 0.01 - 0.005));
            advert.minAmount = 10;
            advert.maxAmount = 5000;
            advert.availableAmount = basePrices[i % 4] * 10;
            advert.ordersCompleted = 50 + i * 10;
            advert.completionRate = 95 + (i % 5);
            advert.avgReleaseTime = 2 + (i % 10);
            advert.isOnline = i % 2 == 0;
            adverts.add(advert);
        }
        return adverts;
    }
    
    public P2POrder createOrder(String advertId, String takerId, double amount) {
        P2POrder order = new P2POrder();
        order.id = "order_" + System.currentTimeMillis();
        order.advertId = advertId;
        order.side = "BUY";
        order.token = "USDT";
        order.fiatCurrency = "USD";
        order.price = 1.0;
        order.amount = amount;
        order.fiatAmount = amount;
        order.status = "PENDING";
        return order;
    }
}

File: mobile_apps/ios_legacy/TigerWallet/Services/Trading/P2PService.swift
//
//  P2PService.swift
//  TigerWallet
//
//  P2P Trading Service - iOS Implementation
//

import Foundation

struct P2PAdvert {
    var id: String
    var userId: String
    var username: String
    var avatar: String
    var side: String
    var token: String
    var fiatCurrency: String
    var paymentMethod: String
    var price: Double
    var minAmount: Double
    var maxAmount: Double
    var availableAmount: Double
    var ordersCompleted: Int
    var completionRate: Double
    var avgReleaseTime: Double
    var isOnline: Bool
}

struct P2POrder {
    var id: String
    var advertId: String
    var side: String
    var token: String
    var fiatCurrency: String
    var paymentMethod: String
    var price: Double
    var amount: Double
    var fiatAmount: Double
    var status: String
}

class P2PService {
    
    func getAdverts(token: String?, fiatCurrency: String?, side: String?) -> [P2PAdvert] {
        let users = ["CryptoTrader1", "BitSeller", "FastTrade", "P2PPro"]
        let avatars = ["🧑‍💼", "👨‍💻", "⚡", "🎯"]
        let payments = ["Bank Transfer", "PayPal", "AliPay", "UPI"]
        let basePrices = [1.0, 43250.0, 2280.0, 312.5]
        
        var adverts: [P2PAdvert] = []
        for i in 0..<users.count {
            let advert = P2PAdvert(
                id: "advert_\(i)",
                userId: "user_\(i)",
                username: users[i],
                avatar: avatars[i],
                side: side ?? (i % 2 == 0 ? "BUY" : "SELL"),
                token: token ?? "USDT",
                fiatCurrency: fiatCurrency ?? "USD",
                paymentMethod: payments[i],
                price: basePrices[i % 4] * (1 + Double.random(in: -0.005...0.005)),
                minAmount: 10,
                maxAmount: 5000,
                availableAmount: basePrices[i % 4] * 10,
                ordersCompleted: 50 + i * 10,
                completionRate: 95 + Double(i % 5),
                avgReleaseTime: 2 + Double(i % 10),
                isOnline: i % 2 == 0
            )
            adverts.append(advert)
        }
        return adverts
    }
    
    func createOrder(advertId: String, takerId: String, amount: Double) -> P2POrder {
        return P2POrder(
            id: "order_\(Int(Date().timeIntervalSince1970 * 1000))",
            advertId: advertId,
            side: "BUY",
            token: "USDT",
            fiatCurrency: "USD",
            paymentMethod: "Bank Transfer",
            price: 1.0,
            amount: amount,
            fiatAmount: amount,
            status: "PENDING"
        )
    }
}
